src/context/WalletProvider/NativeWallet/components/NativeTestPhrase.tsx

```diff
@@ -93,10 +126,9 @@ export const NativeTestPhrase = ({ history, location }: NativeSetupProps) => {
 
   const handleClick = (index: number) => {
     if (index === testState?.correctAnswerIndex) {
-      setInvalidTries([])
       setTestCount(testCount + 1)
     } else {
-      setInvalidTries([...invalidTries, index])
+      shuffleMnemonic().catch(() => setError('walletProvider.shapeShift.create.error'))
```

Comment:
@NeOMakinG curious about two things here:

1. Why do we need to catch, can this actually throw ? (monkey see monkey do, not exactly introduced by yourself, see below)
2. Why do we have a `shuffleMnemonic` effect on l.112 with a dep on itself and do we understand what the double use is doing i.e is there any rationale for it? 

Feels like this user-action-based one should be the only one we ever need?

Reply:
Yeah this part was very bad and working only because of too much rerender

Switched to a more consistant way of working in [e44e559](https://github.com/shapeshift/web/pull/8872/commits/e44e55955ae7e58d8df6f2e6d3618d66913e22de)

- Switch to a ref so we can instantly check the number of step done
- If step done are >= max step, instantly redirect to next step instead of relying on effects
- On load, only execute the shuffling one single time and let the user actions trigger it